src/Xamarin.Android.Build.Tasks/Utilities/ManifestDocument.cs

```diff
@@ -345,20 +345,33 @@ void MergeLibraryManifest (string mergedManifest)
 			var nsResolver = new XmlNamespaceManager (new NameTable ());
 			nsResolver.AddNamespace ("android", androidNs.NamespaceName);
 			var xdoc = XDocument.Load (mergedManifest);
+			var package = xdoc.Root.Attribute ("package")?.Value ?? string.Empty;
+			log.LogDebugMessage ($"Resolved {package} from {mergedManifest}");
 			foreach (var top in xdoc.XPathSelectElements ("/manifest/*")) {
 				var name = top.Attribute (AndroidXmlNamespace.GetName ("name"));
 				var existing = (name != null) ?
 					doc.XPathSelectElement (string.Format ("/manifest/{0}[@android:name='{1}']", top.Name.LocalName, XmlConvert.VerifyNCName (name.Value)), nsResolver) :
 					doc.XPathSelectElement (string.Format ("/manifest/{0}", top.Name.LocalName));
 				if (existing != null)
 					// if there is existing node with the same android:name, then append contents to existing node.
-					existing.Add (top.Nodes ());
+					existing.Add (FixupNameElements (package, top.Nodes ()));
 				else
 					// otherwise, just add to the doc.
-					doc.Root.Add (top);
+					doc.Root.Add (FixupNameElements (package,new XNode [] { top }));
 			}
 		}
 
+		IEnumerable<XNode> FixupNameElements(string packageName, IEnumerable<XNode> nodes)
+		{
+			foreach (var element in nodes.Where ( x => x is XElement).Select (x => x  as XElement)) {
+				foreach (var attr in element.Attributes ()) {
```

Comment:
I'm of the view, we deal with the ones we know about and add as needed.
And yes we should only work on selected "elements"

I'm starting with 

```"activity","service","provider",```

and the attributes you mentioned